libraries/chain/controller.cpp

```diff
@@ -4101,10 +4101,12 @@ struct controller_impl {
       assert(is_proper_savanna_block == b->is_proper_svnn_block());
 
       std::optional<qc_t> qc = verify_basic_block_invariants(id, b, prev);
-
+      log_and_drop_future<void> verify_qc_future;
       if constexpr (is_proper_savanna_block) {
          if (qc) {
-            verify_qc(prev, *qc);
+            verify_qc_future = post_async_task(thread_pool.get_executor(), [this, &qc, &prev] {
+               verify_qc(prev, *qc);
```

Comment:
What happens if there is an exception in `verify_qc()`. Will we just log it and go ahead and create a `block_state`?

Reply:
Yes, `block_state` will be created while the `verify_qc()` is running. If `verify_qc()` does throw then the created `block_state` is thrown away and not added to the fork database.

Reply:
> the created `block_state` is thrown away and not added to the fork database.

What makes this happen?

Reply:
If `verify_qc` throws it is captured by the future. The future rethrows the exception, either here:
https://github.com/AntelopeIO/spring/blob/dbff5a6892bd98256242b26e5a5950690d07c057/libraries/chain/controller.cpp#L4137-L4137
Which will throw out of the method before `forkdb.add()`


Or if another exception is thrown, then the destructor of `log_and_drop_future` logs it. But the other exception that is thrown will also happen before line 4137 and will prevent the `block_state` from being added to the forkdb.

Reply:
Cool, thanks, learned something.